Framework/Algorithms/src/SofQWNormalisedPolygon.cpp

```diff
@@ -411,18 +413,32 @@ RebinnedOutput_sptr SofQWNormalisedPolygon::setUpOutputWorkspace(
   using Kernel::VectorHelper::createAxisFromRebinParams;
 
   HistogramData::BinEdges xAxis(0);
+  auto eHints = std::make_pair<double, double>(std::nan(""), std::nan(""));
   // Create vector to hold the new X axis values
   if (ebinParams.empty()) {
     xAxis = inputWorkspace.binEdges(0);
+  } else if (ebinParams.size() == 1) {
+    eHints = m_EmodeProperties.eBinHints(inputWorkspace);
+    createAxisFromRebinParams(ebinParams, xAxis.mutableRawData(), true, true,
+                              eHints.first, eHints.second);
   } else {
-    static_cast<void>(
-        createAxisFromRebinParams(ebinParams, xAxis.mutableRawData()));
+    createAxisFromRebinParams(ebinParams, xAxis.mutableRawData());
   }
 
   // Create a vector to temporarily hold the vertical ('y') axis and populate
   // that
-  const int yLength = static_cast<int>(
-      VectorHelper::createAxisFromRebinParams(qbinParams, qAxis));
+  int yLength;
+  if (qbinParams.size() == 1) {
+    if (std::isnan(eHints.first)) {
+      eHints = m_EmodeProperties.eBinHints(inputWorkspace);
+    }
+    const auto qHints = m_EmodeProperties.qBinHints(
+        inputWorkspace, eHints.first, eHints.second);
+    yLength = createAxisFromRebinParams(qbinParams, qAxis, true, true,
+                                        qHints.first, qHints.second);
+  } else {
+    yLength = createAxisFromRebinParams(qbinParams, qAxis);
+  }
```

Comment:
This code looks identical to the one above in `SofQW`, can it be extracted into a shared function/method without too much hassle?